### PR TITLE
PSBT Support Added for Blockcore Wallet

### PR DESCRIPTION
**Pull request by tirth2004**
#### Overview
This pull request integrates PSBT (Partially Signed Bitcoin Transactions) signing functionality into the Blockcore wallet 

#### Changes
     - Added PSBT support and routing.
     - Introduced `signPsbt` method to handle PSBT signing requests.
     - Added a new folder `sign-psbt` with component files for the UI and handlers.
     - Implemented `signPsbt`, `prepare`, and `execute` functions in `sign-psbt-handler.ts`.
     - Introduced support for testnet3

#### Documents for clear explanation of the changes
- **Part 1**:
  [Project report mid eval](https://docs.google.com/document/d/17JiT5gG4pW8RUOXc4r-Onc8mfIvEn-3dA4de74y2nXk/edit?usp=sharing)
- **Part 2**:
  [Project report end eval](https://docs.google.com/document/d/1WGEPrXqFruLQgma9700ytwlXXyYuEgMmTu849yXZ0L8/edit?usp=sharing)
- **Test guide**:
  [Test guide](https://docs.google.com/document/d/1we2a26ezo914LcbEMVe4V8Fwl2zK2194si_6yL4Wd8E/edit?usp=sharing)
  
Demo video of PSBT Signing using the new [blazor project](https://github.com/tirth2004/PSBT_Test_App):
https://github.com/user-attachments/assets/4eae95e3-8ded-4e91-9570-cfecc3cb655e

### CODE REVIEW

**Comment by dangershony** [comment]
We are going to have to find some time @sondreb maybe also @miladsoft and myself to review this

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
       - main
   workflow_dispatch:
 
+permissions:
+  contents: write
```
Comment (sondreb) [inline .github/workflows/build.yml:11]: What is this needed for?

File: package.json [hunk 2]
```diff
@@ -46,6 +46,7 @@
     "@angular/platform-browser-dynamic": "14.2.1",
     "@angular/router": "14.2.1",
     "@angular/service-worker": "14.2.1",
+    "@bitcoinerlab/secp256k1": "^1.1.1",
```
Comment (sondreb) [inline package.json:49]: Is this really needed? Please use existing libraries for secp256k1.

**Review by sondreb — CHANGES_REQUESTED** [review]
Thanks for this, please check the comments 👍

File: package.json (hunk 2 above)
Comment (tirth2004) [reply package.json:49]: I was using the existing library , but got some errors. I found this stack article with similar issues:
https://bitcoin.stackexchange.com/questions/118613/webpack-issue-with-tiny-secp256k1-and-ecpair

secp256k1 is suitable replacement, and more widely supported, while also fixed the error.

File: .github/workflows/build.yml (hunk 1 above)
Comment (tirth2004) [reply .github/workflows/build.yml:11]: Couldn't pass the workflow checks on pushing without this

File: package.json (hunk 2 above)
Comment (sondreb) [reply package.json:49]: I see the library is bade on noble-library, so I'll approve this for now.

**Pull request merged**